include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -655,6 +655,18 @@ def TT_DotOp : TT_Op<"dot", [Pure,
 
     let results = (outs TT_FpIntTensor:$d);
 
+
+    let extraClassDeclaration = [{
```

Comment:
nit: please use `DeclareOpInterfaceMethods<DotOpInterface>` and defines these out-of-line in Ops.cpp. Same for the other ops